src/Microsoft.ML.Featurizers/DateTimeTransformer.cs

```diff
@@ -0,0 +1,782 @@
+            _host = Contracts.CheckRef(env, nameof(env)).Register("DateTimeTransformerEstimator");
+
+            _options = options;
+            _options.ColumnsToDrop = _options.ColumnsToDrop == null ? Array.Empty<ColumnsProduced>() : _options.ColumnsToDrop;
+        }
+
+        public DateTimeTransformer Fit(IDataView input)
+        {
+            return new DateTimeTransformer(_host, _options.Source, _options.Prefix, _options.ColumnsToDrop, _options.Country);
+        }
+
+        public SchemaShape GetOutputSchema(SchemaShape inputSchema)
+        {
+            var columns = inputSchema.ToDictionary(x => x.Name);
+
+            foreach (ColumnsProduced column in Enum.GetValues(typeof(ColumnsProduced)))
+                if (_options.ColumnsToDrop == null || !_options.ColumnsToDrop.Contains(column))
+                {
+                    columns[_options.Prefix + column.ToString()] = new SchemaShape.Column(_options.Prefix + column.ToString(), SchemaShape.Column.VectorKind.Scalar,
+                    ColumnTypeExtensions.PrimitiveTypeFromType(column.GetRawColumnType()), false, null);
+                }
+
+            return new SchemaShape(columns.Values);
+        }
+
+        #region Enums
+        public enum ColumnsProduced : byte
+        {
+            Year = 1, Month, Day, Hour, Minute, Second, AmPm, Hour12, DayOfWeek, DayOfQuarter, DayOfYear,
+            WeekOfMonth, QuarterOfYear, HalfOfYear, WeekIso, YearIso, MonthLabel, AmPmLabel, DayOfWeekLabel,
+            HolidayName, IsPaidTimeOff
+        };
+
+        public enum Countries : byte
+        {
+            None = 1,
+            Argentina, Australia, Austria, Belarus, Belgium, Brazil, Canada, Colombia, Croatia, Czech, Denmark,
+            England, Finland, France, Germany, Hungary, India, Ireland, IsleofMan, Italy, Japan, Mexico, Netherlands,
+            NewZealand, NorthernIreland, Norway, Poland, Portugal, Scotland, Slovenia, SouthAfrica, Spain, Sweden, Switzerland,
+            Ukraine, UnitedKingdom, UnitedStates, Wales
+        }
+
+        #endregion
+    }
+
+    public sealed class DateTimeTransformer : RowToRowTransformerBase, IDisposable
+    {
+        #region Class data members
+
+        internal const string Summary = "Splits a date time value into each individual component";
+        internal const string UserName = "DateTime Transform";
+        internal const string ShortName = "DateTimeTransform";
+        internal const string LoadName = "DateTimeTransform";
+        internal const string LoaderSignature = "DateTimeTransform";
+        private DateTimeTypedColumn _column;
+
+        private DateTimeTransformerEstimator.ColumnsProduced[] _columnsToDrop;
+        private byte[] _activeColumnMapping;
+
+        #endregion
+
+        public DateTimeTransformer(IHostEnvironment env, string inputColumnName, string columnPrefix, DateTimeTransformerEstimator.ColumnsProduced[] columnsToDrop, DateTimeTransformerEstimator.Countries country) :
```

Comment:
This constructor shouldn't be public.

Reply:
Changed to internal.

---
In reply to: [353908126](https://github.com/dotnet/machinelearning/pull/4521#discussion_r353908126) [](ancestors = 353908126)